web/util.py

```diff
@@ -24,3 +26,30 @@ def flash_redirect(path: str, message: str | None = None, category="info"):
         flash(message, category)
 
     return redirect(path)
+
+
+def get_printer(config, index=0):
+    """
+    Open configuration object, extract and sanitize printer information, then return it.
+
+    Parameters:
+    config (object): An object containing application configuration.
+    index (int, optional): Index of the printer in the configuration. Default is 0.
+
+    Returns:
+    dict: Dictionary containing sanitized printer information.
+
+    The function logs warnings if the provided index is out of range and defaults it to 0. 
+    It removes 'mqtt_key' and 'p2p_key' from the printer information before returning.
+    """
+    with config.open() as cfg:
+        printers = cfg.printers
+        if index >= len(printers):
+            log.warning(f"Printer number {index} out of range, max printer number is {len(printers)-1}")
```

Comment:
I'm iffy on this particular portion, part of me thinks it isn't necessary here because we shouldn't get to this point with an invalid printer index, but then again, if someone were to re-use this function, they could set a static index that could fall out of range. Thoughts?